src/test/java/org/testeditor/fixture/core/artifacts/TestArtifactRegistryTest.java

```diff
@@ -0,0 +1,282 @@
+                .thenReturn(mockWriter)
+                .thenReturn(mockWriter)
+                .thenThrow(new IOException());
+        } catch (IOException e) {
+            e.printStackTrace();
+            fail("Problem during test setup");
+        }
+        return new FileSystemAccess() {
+            @Override
+            public Writer getBufferedWriter(Path target) throws IOException {
+                return mockWriter;
+            }
+        };
+    }
+
+    private EnvironmentAccess defaultEnvironment = new EnvironmentAccess() {
+        public String get(String key) {
+            switch (key) {
+                case "TE_SUITEID":
+                case "TE_SUITERUNID":
+                case "TE_TESTRUNID":
+                    return "0";
+                default:
+                    return null;
+            }
+        }
+    };
+
+    private EnvironmentAccess emptyEnvironment = new EnvironmentAccess() {
+        public String get(String key) {
+            return null;
+        }
+    };
+
+    private TestArtifact screenshotArtifact = new TestArtifact("screenshot", "screenshots/sampleScreenshot.png");
+    private TestArtifact screencastArtifact = new TestArtifact("screencast", "videos/sampleScreencast.mov");
+
+    private void mockedLogging() {
+        logAppender = mock(Appender.class);
+        logCaptor = ArgumentCaptor.forClass(LogEvent.class);
+        when(logAppender.getName()).thenReturn("MockAppender");
+        when(logAppender.isStarted()).thenReturn(true);
+
+        Logger log4jRootLogger = (Logger) LogManager.getRootLogger();
+        log4jRootLogger.addAppender(logAppender);
+        Configurator.setAllLevels(log4jRootLogger.getName(), WARN);
+    }
+
+    @After
+    public void removeMockedLogAppender() {
+        Logger log4jRootLogger = (Logger) LogManager.getRootLogger();
+        if (logAppender != null && logAppender.getName() != null) {
+            log4jRootLogger.removeAppender(logAppender);
+        }
+    }
+
+    @After
+    public void removeGeneratedFiles() throws IOException {
+        Path targetDir = Paths.get(".testexecution");
+        if (Files.exists(targetDir)) {
+            Files.walk(targetDir).sorted(Comparator.reverseOrder()).map(Path::toFile).forEach(File::delete);
+        }
+    }
+
+    @Test
+    public void getInstanceShouldReturnInitializedRegistry() {
+        // given + when
+        TestArtifactRegistry registry = new TestArtifactRegistry(new FileSystemAccess() {
+        }, defaultEnvironment);
+
+        // then
+        assertThat(registry, notNullValue());
+    }
+
+    @Test
+    public void brokenFileSystemDisablesRegistryButDoesNotThrowAnException() {
+        // given
+        mockedLogging();
+
+        // when
+        new TestArtifactRegistry(brokenFileSystem, defaultEnvironment);
+
+        // then
+        verify(logAppender).append(logCaptor.capture());
+        assertThat(logCaptor.getValue().getMessage().getFormattedMessage(),
```

Comment:
what is with a new line ?

Reply:
the line break is necessary to make CodeStyle happy… And I wanted to avoid breaking in the middle of the string literal.